xtask/src/main.rs

```diff
@@ -97,214 +110,139 @@ fn install_tools() -> Result<()> {
+    uninstall_mdbook_linkcheck()?;
 
     Ok(())
 }
 
-fn uninstall_mdbook_linkcheck() {
+fn uninstall_mdbook_linkcheck() -> Result<()> {
     println!("Uninstalling old mdbook-linkcheck if installed...");
     let output = Command::new(env!("CARGO"))
-        .arg("uninstall")
-        .arg("mdbook-linkcheck")
+        .args(["uninstall", "mdbook-linkcheck"])
         .output()
-        .expect("Failed to execute cargo uninstall mdbook-linkcheck");
+        .context("Failed to execute `cargo uninstall mdbook-linkcheck`")?;
 
     if !output.status.success() {
-        if String::from_utf8_lossy(&output.stderr)
-            .into_owned()
-            .contains("did not match any packages")
-        {
-            println!("mdbook-linkcheck not installed. Continuing...");
-        } else {
-            eprintln!(
-                "An error occurred during uninstallation of mdbook-linkcheck:\n{:#?}",
-                String::from_utf8_lossy(&output.stderr)
-            );
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        // This specific error is OK, it just means the package wasn't installed.
+        if !stderr.contains("did not match any packages") {
+            return Err(anyhow!(
+                "Failed to uninstall `mdbook-linkcheck`.\n--- stderr:\n{stderr}"
+            ));
         }
+        println!("mdbook-linkcheck not installed. Continuing...");
     }
+    Ok(())
 }
 
 fn run_web_tests(dir: Option<PathBuf>) -> Result<()> {
     println!("Running web tests...");
+    let workspace_dir = Path::new(env!("CARGO_WORKSPACE_DIR"));
 
     let absolute_dir = dir.map(|d| d.canonicalize()).transpose()?;
 
     if let Some(d) = &absolute_dir {
         println!("Refreshing slide lists...");
-        let path_to_refresh_slides_script = Path::new("tests")
+        let refresh_slides_script = Path::new("tests")
             .join("src")
             .join("slides")
             .join("create-slide.list.sh");
-        let status = Command::new(path_to_refresh_slides_script)
-            .current_dir(Path::new(env!("CARGO_WORKSPACE_DIR")))
-            .arg(d)
-            .status()
-            .expect("Failed to execute create-slide.list.sh");
-
-        if !status.success() {
-            let error_message = format!(
-                "Command 'cargo xtask web-tests' exited with status code: {}",
-                status.code().unwrap()
-            );
-            return Err(anyhow!(error_message));
-        }
+        let mut cmd = Command::new(&refresh_slides_script);
+        cmd.current_dir(workspace_dir).arg(d);
+        run_command(&mut cmd)?;
     }
 
-    let path_to_tests_dir = Path::new(env!("CARGO_WORKSPACE_DIR")).join("tests");
-    let mut command = Command::new("npm");
-    command.current_dir(path_to_tests_dir.to_str().unwrap());
-    command.arg("test");
+    let tests_dir = workspace_dir.join("tests");
+    let mut cmd = Command::new("npm");
+    cmd.current_dir(tests_dir).arg("test");
```

Comment:
Something like this should work, especially if `run_command` would take ownership of the `Command` passed in:

```suggestion
    let cmd = Command::new("npm");
        .current_dir(tests_dir).arg("test");
```